lib/url-decorator.jsx

```diff
@@ -0,0 +1,51 @@
+import React from 'react';
+import { CompositeDecorator } from 'draft-js';
+import urlRegex from 'url-regex';
+import normalizeUrl from 'normalize-url';
+import { omit } from 'lodash';
+import { viewExternalUrl } from './utils/url-utils';
+
+const urlRe = urlRegex();
+const markdownLinkRe = /\(|\]:\s$/m;
+
+export function normalizeUrlsInMarkdownString( str ) {
+	return str.replace( urlRe, ( match, offset ) => {
+		var normalized = normalizeUrl( match );
+		if ( markdownLinkRe.test( str.slice( 0, offset ) ) ) {
+			return normalized;
+		}
+		// replace url with markdown link only if it's not already part of one
+		return `[${match}](${normalized})`;
+	} );
+}
+
+export const urlDecorator = new CompositeDecorator( [
+	{
+		strategy: function( contentBlock, callback ) {
+			const text = contentBlock.getText();
+			let matchArr, start;
+			while ( ( matchArr = urlRe.exec( text ) ) !== null ) {
+				start = matchArr.index;
+				callback( start, start + matchArr[0].length );
+			}
+		},
+
+		component: React.createClass( {
+			onClick: function( event ) {
+				event.preventDefault();
+				viewExternalUrl( normalizeUrl( this.props.decoratedText ) );
+			},
+
+			render: function() {
+				return (
+					<a
+						{...omit( this.props, 'entityKey', 'offsetKey', 'decoratedText' )}
```

Comment:
actually there's no reason to pass any props at all I think, so I've just removed that line